packages/core/src/components/Timer.tsx

```diff
@@ -0,0 +1,139 @@
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  useImperativeHandle,
+  forwardRef,
+} from "react";
+import { Text, StyleSheet, TextStyle } from "react-native";
+
+interface TimerProps {
+  style?: TextStyle;
+  initialTime?: number;
+  updateInterval?: number;
+  format?: "ss" | "mm:ss" | "hh:mm:ss" | "ss:ms" | "mm:ss:ms" | "hh:mm:ss:ms";
+  onTimerChange?: (time: number) => void;
+  onTimerEnd?: () => void;
+  countDirection?: "up" | "down";
+}
+
+export interface TimerHandle {
+  start: () => void;
+  stop: () => void;
+  reset: (newTime?: number) => void;
+}
+
+const Timer = forwardRef<TimerHandle, TimerProps>(
+  (
+    {
+      style,
+      initialTime,
+      updateInterval = 1000,
+      format = "mm:ss",
+      onTimerChange,
+      onTimerEnd,
+      countDirection = "up",
+    },
+    ref
+  ) => {
+    const defaultInitialTime = countDirection === "up" ? 0 : 100000;
+    const [time, setTime] = useState(initialTime ?? defaultInitialTime);
+    const timerRef = useRef<NodeJS.Timeout | null>(null);
+
+    useEffect(() => {
+      onTimerChange?.(time);
+    }, [time, onTimerChange]);
+
+    useImperativeHandle(ref, () => ({
+      start: startTimer,
+      stop: stopTimer,
+      reset: resetTimer,
+    }));
+
+    const startTimer = () => {
+      if (timerRef.current) return;
+      timerRef.current = setInterval(() => {
+        setTime((prevTime) => {
+          const newTime =
+            countDirection === "up"
+              ? prevTime + updateInterval
+              : prevTime - updateInterval;
+          if (newTime <= 0 && countDirection === "down") {
```

Comment:
So if the direction is `"up"`, the timer never ends? Maybe we can have a prop `timerEndTime` or something similar that tells us when a timer should be considered completed